assets/js/components/Dashboard/NewPaMessagePage.tsx

```diff
@@ -86,17 +113,56 @@ const NewPaMessagePage = () => {
         <div className="new-pa-message-page__header">New PA/ESS message</div>
         <Container fluid>
           <Row md="auto" className="align-items-center">
-            <Button variant="link" className="pr-0">
-              Associate with alert
-            </Button>
-            (Optional)
-          </Row>
-          <Row>
-            <div className="new-pa-message-page__associate-alert-subtext">
-              Linking will allow you to share end time with alert, and import
-              location and message.
-            </div>
+            {associatedAlert.id ? (
+              <div className="associated-alert-header">
+                Associated Alert: Alert ID {associatedAlert.id}
+                <Button
+                  variant="link"
+                  onClick={() => navigate("associate-alert")}
+                >
+                  Edit
+                </Button>
+                <Button
+                  variant="link"
+                  onClick={() => {
+                    setVisualText("");
+                    setPhoneticText("");
+                    setEndDate(now.format("L"));
+                    setEndTime(now.add(1, "hour").format("HH:mm"));
+                    dispatch({
+                      type: "SET_ASSOCIATED_ALERT",
+                      associatedAlert: {} as AlertModel,
+                      endWithEffectPeriod: false,
+                      importLocations: false,
+                      importMessage: false,
+                    });
+                  }}
+                >
+                  Clear
+                </Button>
+                {formatActivePeriod(associatedAlert.active_period)}
+              </div>
+            ) : (
+              <>
+                <Button
+                  variant="link"
+                  className="pr-0"
+                  onClick={() => navigate("associate-alert")}
+                >
+                  Associate with alert
+                </Button>
+                (Optional)
+              </>
+            )}
           </Row>
+          {!associatedAlert.id && (
+            <Row>
+              <div className="new-pa-message-page__associate-alert-subtext">
+                Linking will allow you to share end time with alert, and import
+                location and message.
+              </div>
+            </Row>
+          )}
```

Comment:
Can this be pulled out into its own component? Some of this is getting a bit hard to follow with all the different conditionals.